### PR TITLE
kv: improve test coverage above 80%

### PR DESCRIPTION
**Pull request by tiancaiamao**
For https://github.com/pingcap/tidb/issues/1613

### CODE REVIEW

File: kv/buffer_store_test.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	key := Key("key")
+	value, err := bs.Get(key)
+	c.Check(err, NotNil)
+
+	err = bs.Set(key, []byte("value"))
+	c.Check(err, IsNil)
+
+	value, err = bs.Get(key)
+	c.Check(err, IsNil)
+	c.Check(0, Equals, bytes.Compare(value, []byte("value")))
```
Comment (coocood) [inline kv/buffer_store_test.go]: swap 0 and `bytes.Compare...`

File: kv/mock_test.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	storage.UUID()
+	version, err := storage.CurrentVersion()
+	c.Check(err, IsNil)
+	snapshot, err := storage.GetSnapshot(version)
+	snapshot.BatchGet([]Key{Key("abc"), Key("def")})
+	c.Check(err, IsNil)
+
+	transaction, err := storage.Begin()
+	c.Check(err, IsNil)
+	transaction.LockKeys(Key("lock"))
```
Comment (shenli) [inline kv/mock_test.go]: Should we check the results in the following code?

**Comment by tiancaiamao** [comment]
It's in a dilemma put mock code just in test package or not:
On one hand, test the mocked code for the sake of coverage is meaningless, so it should be moved to xxx_test.go -- xxx_test.go doesn't count for coverage.
On the other hand, if move mock code to xxx_test.go, other package can't use that code, build would fail.

**Comment by shenli** [comment]
LGTM

**Comment by shenli** [comment]
@coocood PTAL

**Comment by coocood** [comment]
LGTM

**Pull request merged**